applications/StructuralMechanicsApplication/tests/test_truss_adjoint_element_3d2n.py

```diff
@@ -0,0 +1,288 @@
+        for j in range(6):
+            FDPseudoLoadMatrix[row_index,j] = (RHSPerturbed[j] - RHSUnperturbed[j]) / delta
+        row_index = row_index + 1
+
+    return FDPseudoLoadMatrix
+
+def assert_matrix_almost_equal(matrix1, matrix2, prec=7):
+    KratosUnittest.TestCase().assertEqual(matrix1.Size1(), matrix2.Size1())
+    KratosUnittest.TestCase().assertEqual(matrix1.Size2(), matrix2.Size2())
+    for i in range(matrix1.Size1()):
+        for j in range(matrix1.Size2()):
+            KratosUnittest.TestCase().assertAlmostEqual(matrix1[i,j], matrix2[i,j], prec)
+
+class TestTrussLinearAdjointElement(KratosUnittest.TestCase):
+
+    def setUp(self):
+        # create test model part
+        dim=3
+        self.model_part = KratosMultiphysics.ModelPart("test")
+        self.model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE,dim)
+        add_variables(self.model_part)
+        self.model_part.CreateNewNode(1, 0.0, 0.0, 0.0)
+        self.model_part.CreateNewNode(2, 1.0, 1.0, 0.3)
+        apply_material_properties(self.model_part,dim)
+        prop = self.model_part.GetProperties()[1]
+
+        self.model_part.CreateNewElement("TrussLinearElement3D2N", 1, [1, 2], prop)
+        StructuralMechanicsApplication.ReplaceElementsAndConditionsForAdjointProblemProcess(
+            self.model_part).Execute()
+        self.adjoint_truss_element = self.model_part.GetElement(1)
+
+        self.model_part.CreateNewElement("TrussLinearElement3D2N", 2, [1, 2], prop)
+        self.truss_element = self.model_part.GetElement(2)
+
+        self.model_part.Nodes[1].SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X, 0, 0.014725)
+        self.model_part.Nodes[1].SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y, 0, 0.001200)
+        self.model_part.Nodes[1].SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z, 0, 0.0725715)
+        self.model_part.Nodes[2].SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X, 0, 0.019735)
+        self.model_part.Nodes[2].SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y, 0, 0.002400)
+        self.model_part.Nodes[2].SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z, 0, 0.377976)
+
+        self.truss_element.Initialize()
+        self.adjoint_truss_element.Initialize()
+
+
+    def test_CalculateSensitivityMatrix_Property(self):
+        # Pertubation measure
+        h = 0.00001
+        A_initial = self.model_part.GetProperties()[1][StructuralMechanicsApplication.CROSS_AREA]
+        delta = A_initial * h
+
+        # Create perturbed elements
+        perturbed_mp = create_property_perturbed_elements(self.model_part,delta,"TrussLinearElement3D2N")
+        
+        # Derive RHS by finite differences
+        FDPseudoLoadMatrix = FD_calculate_sensitivity_matrix(self.truss_element, self.model_part, perturbed_mp, delta)  
+        
+        # Pseudo-load computation by adjoint element
+        PseudoLoadMatrix = KratosMultiphysics.Matrix(1,6)
```

Comment:
is it necessary to define the size already here?
There should be a resize inside the function

Reply:
You are right, sizing is not necessary.